lib/BasicAnalyses/GeneratedCodeBasicInfo.cpp

```diff
@@ -141,3 +135,34 @@ GeneratedCodeBasicInfo::getSuccessors(BasicBlock *BB) const {
 
   return Result;
 }
+
+AnalysisKey GeneratedCodeBasicInfoAnalysis::Key;
+GeneratedCodeBasicInfo
+GeneratedCodeBasicInfoAnalysis::run(Module &M,
+                                    ModuleAnalysisManager &MAM) {
+  GeneratedCodeBasicInfo GCBI;
+  GCBI.run(M);
+  return GCBI;
+}
+
+GeneratedCodeBasicInfo
+GeneratedCodeBasicInfoAnalysis::run(Function &F,
+                                    FunctionAnalysisManager &FAM) {
+  GeneratedCodeBasicInfo GCBI;
+  GCBI.run(*F.getParent());
+  return GCBI;
+}
+
+bool GeneratedCodeBasicInfoWrapperPass::runOnModule(Module &M) {
+  GCBI.reset(new GeneratedCodeBasicInfo());
+  GCBI->run(M);
+  return false;
+}
+
+void GeneratedCodeBasicInfoWrapperPass::releaseMemory() {
+  GCBI->releaseMemory();
+}
```

Comment:
In `GeneratedCodeBasicInfoWrapperPass::releaseMemory` just call `GCBI.reset(new GeneratedCodeBasicInfo());`.
I don't think we need to implement `releaseMemory` for `GeneratedCodeBasicInfo` at this point. Even because `.clear` doesn't actually release memory, while `reset`ting the `unique_ptr` does.

Reply:
Well, `releaseMemory` in GCBI is kind of unneeded, but it doesn't harm calling `clear` and `reset`. Same for `GCBI.reset()`, which is automatically done, but yeah, adding it is harmless.

Reply:
Harmless is nice way of saying "useless".
There's no point in `releaseMemory` in GCBI, it's a pattern common in old passes, not in analysis results.
Just `GCBI.reset` and drop `GeneratedCodeBasicInfo::releaseMemory`.